src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/leaked-passwords/LeakedPasswordsLayout.tsx

```diff
@@ -18,76 +19,192 @@ import {
+  }
+}
+/* c8 ignore stop */
+
+export function LeakedPasswordsLayout(props: LeakedPasswordsLayoutProps) {
+  // This will be always the initial props passed from the page.
+  // They were fetched from the db.
+  const [subscriberBreaches, setSubscriberBreaches] = useState(
     props.data.subscriberBreaches,
+  );
+  const guidedExperienceBreaches = getGuidedExperienceBreaches(
+    subscriberBreaches,
     props.subscriberEmails,
   );
-
-  const pageData = getLeakedPasswords({
+  const stepMap: Record<LeakedPasswordsTypes, StepLink["id"]> = {
+    passwords: "LeakedPasswordsPassword",
+    "security-question": "LeakedPasswordsSecurityQuestion",
+  };
+  const l10n = useL10n();
+  const [emailAffected, setEmailAffected] = useState<string>();
+  const [pageDataContent, setPageDataContent] = useState<LeakedPassword>({
     dataType: props.type,
     breaches: guidedExperienceBreaches,
     l10n: l10n,
+    emailAffected: emailAffected ?? "",
   });
 
-  // The non-null assertion here should be safe since we already did this check
-  // in `./[type]/page.tsx`:
-  const { title, illustration, content } = pageData!;
+  const pageData = getLeakedPasswords(pageDataContent);
+  const unresolvedPasswordBreachContent =
+    pageData.unresolvedPasswordBreachContent;
+  const unresolvedPasswordBreach =
+    pageData.unresolvedPasswordBreach ??
+    pageData.unresolvedSecurityQuestionsBreach;
+  const resolvedDataClassName =
+    props.type === "passwords" ? "passwords" : "security-questions-and-answers";
 
-  return (
-    <FixView
-      subscriberEmails={props.subscriberEmails}
-      data={props.data}
-      nextStepHref={getNextGuidedStep(props.data, stepMap[props.type]).href}
-      currentSection="leaked-passwords"
-    >
-      <ResolutionContainer
-        type="leakedPasswords"
-        title={title}
-        illustration={illustration}
-        cta={
-          <>
-            <Button
-              variant="primary"
-              small
-              // TODO: Add test once MNTOR-1700 logic is added
-              /* c8 ignore next 3 */
-              onPress={() => {
-                // TODO: MNTOR-1700 Add routing logic
-              }}
-              autoFocus={true}
-            >
-              {l10n.getString("leaked-passwords-mark-as-fixed")}
-            </Button>
-            <Link
-              // TODO: Add test once MNTOR-1700 logic is added
-              href="/"
-            >
-              {l10n.getString("leaked-passwords-skip")}
-            </Link>
-          </>
+  useEffect(() => {
+    if (emailAffected) {
+      const newPageData = {
+        dataType: props.type,
+        breaches: guidedExperienceBreaches,
+        l10n: l10n,
+        emailAffected: emailAffected,
+      };
+      setPageDataContent(newPageData);
+    }
+    // This should only run if emailAffected changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [emailAffected]);
+
+  useEffect(() => {
```

Comment:
It seems like this `useEffect` is not needed. Instead of storing `emailAffected` in a state we should be able to just use it as a variable?